controllers/alert_controller.go

```diff
@@ -153,3 +176,24 @@ func (r *AlertReconciler) validate(ctx context.Context, alert *v1beta1.Alert) er
 
 	return nil
 }
+
+func (r *AlertReconciler) requestsForProviderChange(o client.Object) []reconcile.Request {
```

Comment:
great! don't you think though that the name would be better like `reconcileProviderDependentObjects`